boards/stm32f3discovery/src/main.rs

```diff
@@ -0,0 +1,429 @@
+    // Create the debugger object that handles calls to `debug!()`.
+    components::debug_writer::DebugWriterComponent::new(uart_mux).finalize(());
+
+    // // Setup the process inspection console
+    // let process_console_uart = static_init!(UartDevice, UartDevice::new(mux_uart, true));
+    // process_console_uart.setup();
+    // pub struct ProcessConsoleCapability;
+    // unsafe impl capabilities::ProcessManagementCapability for ProcessConsoleCapability {}
+    // let process_console = static_init!(
+    //     capsules::process_console::ProcessConsole<'static, ProcessConsoleCapability>,
+    //     capsules::process_console::ProcessConsole::new(
+    //         process_console_uart,
+    //         &mut capsules::process_console::WRITE_BUF,
+    //         &mut capsules::process_console::READ_BUF,
+    //         &mut capsules::process_console::COMMAND_BUF,
+    //         board_kernel,
+    //         ProcessConsoleCapability,
+    //     )
+    // );
+    // hil::uart::Transmit::set_transmit_client(process_console_uart, process_console);
+    // hil::uart::Receive::set_receive_client(process_console_uart, process_console);
+    // process_console.start();
+
+    // LEDs
+
+    // Clock to Port E is enabled in `set_pin_primary_functions()`
+    let led_pins = static_init!(
+        [(
+            &'static dyn kernel::hil::gpio::Pin,
+            capsules::led::ActivationMode
+        ); NUM_LEDS],
+        [
+            (
+                stm32f3xx::gpio::PinId::PE09.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE08.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE10.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE15.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE11.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE14.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE12.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+            (
+                stm32f3xx::gpio::PinId::PE13.get_pin().as_ref().unwrap(),
+                capsules::led::ActivationMode::ActiveHigh
+            ),
+        ]
+    );
+    let led = static_init!(
+        capsules::led::LED<'static>,
+        capsules::led::LED::new(&led_pins[..])
+    );
```

Comment:
Please update to the component.

Reply:
I'm not sure I understand, this is how stm32f4xx is done. Probably I am missing something.

Reply:
It looks like the other stm32 boards have not been fully updated to use components (in praticular, the LED driver and alarm haven't but some others have been).

You can look at other boards (Hail and imix) for how this _should_ be done, and it's maybe worth updating the nucleo boards in a separate PR.

Reply:
Got it already, will fix for discovery and nucleo boards.

Reply:
Done